internal/testing/testdata/testdata.go

```diff
@@ -420,200 +423,81 @@ var (
-			},
-		),
-	}
-
-	cdxTzdataPack = assembler.PackageNode{
-		Name:    "tzdata",
-		Digest:  nil,
-		Version: "2021a-1+deb11u6",
-		Purl:    "pkg:deb/debian/tzdata@2021a-1+deb11u6?arch=all&distro=debian-11",
-		CPEs: []string{
-			"cpe:2.3:a:tzdata:tzdata:2021a-1\\+deb11u6:*:*:*:*:*:*:*"},
-		NodeData: *assembler.NewObjectMetadata(
-			processor.SourceInformation{
-				Collector: "TestCollector",
-				Source:    "TestSource",
-			},
-		),
-	}
+	cdxTzdataPack, _ = asmhelpers.PurlToPkg("pkg:deb/debian/tzdata@2021a-1+deb11u6?arch=all&distro=debian-11")
 
-	cdxNetbasePack = assembler.PackageNode{
-		Name:    "netbase",
-		Digest:  nil,
-		Version: "6.3",
-		Purl:    "pkg:deb/debian/netbase@6.3?arch=all&distro=debian-11",
-		CPEs: []string{
-			"cpe:2.3:a:netbase:netbase:6.3:*:*:*:*:*:*:*"},
-		NodeData: *assembler.NewObjectMetadata(
-			processor.SourceInformation{
-				Collector: "TestCollector",
-				Source:    "TestSource",
-			},
-		),
-	}
+	cdxNetbasePack, _ = asmhelpers.PurlToPkg("pkg:deb/debian/netbase@6.3?arch=all&distro=debian-11")
 
-	cdxBasefilesPack = assembler.PackageNode{
-		Name:    "base-files",
-		Digest:  nil,
-		Version: "11.1+deb11u5",
-		Purl:    "pkg:deb/debian/base-files@11.1+deb11u5?arch=amd64&distro=debian-11",
-		CPEs: []string{
-			"cpe:2.3:a:base-files:base-files:11.1\\+deb11u5:*:*:*:*:*:*:*"},
-		NodeData: *assembler.NewObjectMetadata(
-			processor.SourceInformation{
-				Collector: "TestCollector",
-				Source:    "TestSource",
-			},
-		),
-	}
+	cdxBasefilesPack, _ = asmhelpers.PurlToPkg("pkg:deb/debian/base-files@11.1+deb11u5?arch=amd64&distro=debian-11")
 
-	CycloneDXNodes = []assembler.GuacNode{cdxTopLevelPack, cdxBasefilesPack, cdxNetbasePack, cdxTzdataPack}
-	CyloneDXEdges  = []assembler.GuacEdge{
-		assembler.DependsOnEdge{
-			PackageDependency: cdxBasefilesPack,
-			PackageNode:       cdxTopLevelPack,
+	CdxDeps = []assembler.IsDependencyIngest{
+		{
+			Pkg:          cdxTopLevelPack,
+			DepPkg:       cdxBasefilesPack,
+			IsDependency: isDepJustifyTopPkg,
 		},
-		assembler.DependsOnEdge{
-			PackageDependency: cdxNetbasePack,
-			PackageNode:       cdxTopLevelPack,
+		{
+			Pkg:          cdxTopLevelPack,
+			DepPkg:       cdxNetbasePack,
+			IsDependency: isDepJustifyTopPkg,
 		},
-		assembler.DependsOnEdge{
-			PackageDependency: cdxTzdataPack,
-			PackageNode:       cdxTopLevelPack,
+		{
+			Pkg:          cdxTopLevelPack,
+			DepPkg:       cdxTzdataPack,
+			IsDependency: isDepJustifyTopPkg,
 		},
 	}
 
-	// CycloneDX Testdata with package dependencies
-	cdxTopQuarkusPack = assembler.PackageNode{
-		Name:    "getting-started",
-		Version: "1.0.0-SNAPSHOT",
-		Purl:    "pkg:maven/org.acme/getting-started@1.0.0-SNAPSHOT?type=jar",
-		Tags:    []string{"library"},
-		CPEs:    nil,
-		NodeData: *assembler.NewObjectMetadata(
-			processor.SourceInformation{
-				Collector: "TestCollector",
-				Source:    "TestSource",
-			},
-		),
+	CdxIngestionPredicates = assembler.IngestPredicates{
+		IsDependency: CdxDeps,
```

Comment:
+ the IsOccurrence predicate (from earlier)

Reply:
Artifacts will be implemented in the next iteration.